code/game/objects/structures/tables_racks.dm

```diff
@@ -591,10 +591,21 @@
 	var/mob/living/carbon/M = locate(/mob/living/carbon) in loc
 	if(M)
 		if(M.resting)
-			patient = M
+			set_patient(M)
 	else
-		patient = null
+		set_patient(null)
 
+/obj/structure/table/optable/proc/set_patient(new_patient)
+	if(patient)
```

Comment:
This seems to be a pattern you repeat a lot. Can we make it a proc? As in...
```dm
something_something(patient, .proc/patient_deleted, new_patient)
patient = new_patient
```

Reply:
Would that even look better?

Reply:
you'd need to pass in var name, and it'd be a real mess. Kinda tempting, but I'm worried about the results of doing it